File: app/src/main/java/com/sorcerer/sorcery/iconpack/utils/StringUtil.java
package com.sorcerer.sorcery.iconpack.utils;

import timber.log.Timber;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/4/26
 */
public class StringUtil {

    public static String handleLongXmlString(String s) {
        String res = s.replace("|", "\n");
        res = res.replace("#", "    ");
        return res;
    }

    public static String componentInfoToPackageName(String componentInfo) {
        try {
            return componentInfo.split("/")[0].split("\\{")[1];
        } catch (Exception e) {
            Timber.e(e);
        }
        return null;
    }

}

File: app/src/main/java/com/sorcerer/sorcery/iconpack/utils/StringUtil.kt
package com.sorcerer.sorcery.iconpack.utils

import timber.log.Timber

/**
 * @description:
 * *
 * @author: Sorcerer
 * *
 * @date: 2016/4/26
 */
object StringUtil {
    fun isMail(mail: String): Boolean {
        return mail.matches(
                "^([a-zA-Z0-9_-])+@([a-zA-Z0-9_-])+((\\.[a-zA-Z0_9_-]{2,3}){1,2})$".toRegex())
    }

    fun handleLongXmlString(s: String): String {
        var res = s.replace("|", "\n")
        res = res.replace("#", "    ")
        return res
    }

    fun componentInfoToPackageName(componentInfo: String): String? {
        try {
            return componentInfo.split(
                    "/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0].split(
                    "\\{".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[1]
        } catch (e: Exception) {
            Timber.e(e)
        }

        return null
    }

}
